src/layers/basic.jl

```diff
@@ -692,3 +692,126 @@ end
 function Base.show(io::IO, m::Embedding)
   print(io, "Embedding(", size(m.weight, 2), " => ", size(m.weight, 1), ")")
 end
+
+"""
+    EmbeddingBag(in => out, reduction=mean; init=randn)
+
+A lookup table that stores embeddings of dimension `out` for a vocabulary of size 
+`in`. Similar to [`Embedding`](@ref) but can take multiple inputs in a "bag". The
+embeddings of these are then reduced to a single embedding based on `reduction`.
+Typically, `reduction` is `mean`, `sum`, or `maximum`. 
+
+This layer is often used to store word embeddings and retrieve them using indices. 
+The inputs can take several forms:
+  - A scalar := single bag with a single item
+  - A vector := single bag with multiple items
+  - A matrix := multiple bags with multiple items (each column is a bag)
+  - A vector of vectors := multiple bags with multiple items (each vector is a bag)
+  - An input vector and offset vector := Explained below.
+
+  The `input`/`offset` input type is similar to PyTorch's implementation. `input` should be
+  a vector of class indices and `offset` should be a vector representing the starting index of a bag in the `inputs` vector. The first element of `offsets` must be `1`, and `offsets` should
+  be monotonically increasing, but the second condition is not checked.
+
+  This format is useful for dealing with flattened representations of "ragged" tensors. E.g., if you have a flat vector of class labels that need to be grouped in a non-uniform way. However, under the hood, it is just syntactic sugar for the vector-of-vectors input style.
+
+  For example, the `input`/`offset` pair `[1, 2, 3, 4, 5, 6, 7, 8, 9, 10]`/`[1, 5, 6, 8]`
+  is equivalent to the bags `[[1, 2, 3, 4], [5], [6, 7], [8, 9, 10]]`, since the first bag starts at index `1` and goes up to index `5`, non-inclusive. The next bag starts at index `5` and goes up to index `6`, non-inclusive, etc. Below is another example usage.
+
+# Examples
+```jldoctest
+julia> vocab_size, embed_size = 1000, 4;
+
+julia> model = Flux.EmbeddingBag(vocab_size => embed_size)
+EmbeddingBag(1000 => 4)  # 4_000 parameters
+
+julia> bags = [[1, 200, 25, 789], [2, 5, 10, 999]];
+
+julia> bags_mtx = [1 2; 200 5; 25 10; 789 999];
+
+julia> model(bags) |> summary
+"4×2 Matrix{Float32}"
+
+julia> model(bags) ≈ model(bags_mtx)
+true
+```
+
+```jldoctest
+julia> vocab_size, embed_size = 10, 8;
+
+julia> model = Flux.EmbeddingBag(vocab_size => embed_size)
+EmbeddingBag(10 => 8)  # 80 parameters
+
+julia> scalar_bag = 5 # just a single bag of one item
+5
+
+julia> model(scalar_bag);
+
+julia> single_bag = [1, 2, 2, 4]; # one bag several items
+
+julia> model(single_bag) |> summary
+"8-element Vector{Float32}"
```

Comment:
There are 2 example blocks, maybe not both needed? 

And maybe this can be compacted a bit, some things don't need to be given variable names:
```suggestion
julia> model = Flux.EmbeddingBag(26 => 10, mean)  # vocab size 26, output 10 dims
EmbeddingBag(26 => 10)  # 260 parameters

julia> model(5) |> summary  # one item
"10-element Vector{Float32}"

julia> model([1, 2, 2, 4]) |> summary  # one bag
"10-element Vector{Float32}"

julia> model([1, 2, 4]) ≈ mean(model.([1, 2, 3]))
```

Reply:
I kept one block specifically for the `input`/`offsets` example, since that is complex and one to demonstrate each of the input types. I will make it more concise though.

Reply:
Yes, that (if it stays) could happily be a second block below.